### DISCUSSION TITLE
Is it possible to type the error for useQuery while inferring the data type?

### DISCUSSION THREAD

**Discussion by HemalR** [TanStack/query#4268] [Q&A] [answered]
The error type can be explicitly defined (https://tanstack.com/query/v4/docs/typescript#typing-the-error-field) by passing the types for both `data` and `error`.

Is there a way to do this while continuing to infer the type for `data`?

Thanks in advance!

**Comment by TkDodo** [comment, marked as answer]
no because TypeScript has no way to do partial type inference. see this open issue in TS:
- https://github.com/microsoft/TypeScript/issues/26242

However, we also have some slight improvements planned here to default `TError` to `Error` in v5. Have a look here:
- https://github.com/TanStack/query/discussions/4252

**Reply by HemalR** [reply]
Awesome, thanks! 👍
